### PR TITLE
Provide option to set theme based on OS theme preference.#5932

### PR DESCRIPTION
**Pull request by yogeshmahajan-1903**

### CODE REVIEW

File: web/pgadmin/misc/__init__.py [hunk 1]
```diff
@@ -84,11 +83,17 @@ def register_preferences(self):
             options=theme_options,
             control_props={
                 'allowClear': False,
+                'creatable': False,
             },
+        )
+
+        self.preference.register(
```
Comment (adityatoshniwal) [inline web/pgadmin/misc/__init__.py:90]: there is no need for a new preference. You could add one more drop-down option in existing themes - `System`

File: web/pgadmin/misc/themes/__init__.py [hunk 2]
```diff
@@ -5,9 +5,9 @@
 
 def get_all_themes():
     all_themes = {
-        "standard": {
-            "disp_name": "Standard",
-            "preview_img": "standard_preview.png"
+        "light": {
```
Comment (adityatoshniwal) [inline web/pgadmin/misc/themes/__init__.py:8]: you'll also have to change the default value for theme preference.

File: web/pgadmin/preferences/static/js/components/PreferencesComponent.jsx [hunk 3]
```diff
@@ -292,6 +292,16 @@ export default function PreferencesComponent({ ...props }) {
           opt.selected = false;
         }
       });
+
+      let sync_with_os_id = null;
+      subNode.preferences.forEach((_el) => {
```
Comment (adityatoshniwal) [inline web/pgadmin/preferences/static/js/components/PreferencesComponent.jsx:297]: you can remove this once you the same `theme` preference.

**Review by adityatoshniwal — CHANGES_REQUESTED** [review]

File: web/pgadmin/misc/__init__.py (hunk 1 above)
Comment (yogeshmahajan-1903) [reply web/pgadmin/misc/__init__.py:90]: Generally, sync with os option is given separately for theme selection for example - pycharm. Providing option this way will be more noticeable than in dropdown.

File: web/pgadmin/misc/themes/__init__.py (hunk 2 above)
Comment (yogeshmahajan-1903) [reply web/pgadmin/misc/themes/__init__.py:8]: Done

File: web/pgadmin/misc/__init__.py (hunk 1 above)
Comment (adityatoshniwal) [reply web/pgadmin/misc/__init__.py:90]: Slack, Chrome has different way of doing it. Even VS code.
The point is - having 2 variables to decide a theme can be more confusing. It should be either one.

image

image

Comment (adityatoshniwal) [reply web/pgadmin/misc/__init__.py:90]: Firefox. Most apps do not have a separate option.

image

Comment (yogeshmahajan-1903) [reply web/pgadmin/misc/__init__.py]: Done.

File: web/pgadmin/preferences/static/js/components/PreferencesComponent.jsx (hunk 3 above)
Comment (yogeshmahajan-1903) [reply web/pgadmin/preferences/static/js/components/PreferencesComponent.jsx]: Done.

**Pull request merged**